### PR TITLE
Fix error message when creating `Settings.toml` without access

### PR DESCRIPTION
**Pull request by parkavi11**
Fixes https://github.com/ballerina-platform/ballerina-lang/issues/29716

## Check List 
- [ ] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [ ] Added necessary tests
  - [ ] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: compiler/ballerina-lang/src/main/java/io/ballerina/projects/internal/environment/BallerinaUserHome.java [hunk 1]
```diff
@@ -36,6 +37,8 @@ private BallerinaUserHome(Environment environment, Path ballerinaUserHomeDirPath
                 .resolve(ProjectConstants.CENTRAL_REPOSITORY_CACHE_NAME);
         try {
             Files.createDirectories(remotePackageRepositoryPath);
+        } catch (AccessDeniedException ae) {
+            throw new ProjectException("permission denied: " + ae.getMessage());
```
Comment (pramodya1994) [inline compiler/ballerina-lang/src/main/java/io/ballerina/projects/internal/environment/BallerinaUserHome.java:41]:
```
Failed to create /home/user/.ballerina/Settings.toml: permission denied
```
Shall we give the error with the path? How this error will be printed now? Can you add that to the PR as well?

File: compiler/ballerina-lang/src/main/java/io/ballerina/projects/internal/environment/BallerinaUserHome.java [hunk 2]
```diff
@@ -32,10 +33,15 @@
 
     private BallerinaUserHome(Environment environment, Path ballerinaUserHomeDirPath) {
         this.ballerinaUserHomeDirPath = ballerinaUserHomeDirPath;
+        Path settingsTomlPath = Paths.get(String.valueOf(ballerinaUserHomeDirPath),
+                ProjectConstants.SETTINGS_FILE_NAME);
         Path remotePackageRepositoryPath = ballerinaUserHomeDirPath.resolve(ProjectConstants.REPOSITORIES_DIR)
                 .resolve(ProjectConstants.CENTRAL_REPOSITORY_CACHE_NAME);
         try {
             Files.createDirectories(remotePackageRepositoryPath);
+        } catch (AccessDeniedException ae) {
```
Comment (azinneera) [inline compiler/ballerina-lang/src/main/java/io/ballerina/projects/internal/environment/BallerinaUserHome.java:42]: We are not trying to create the Settings.toml here. We are only creating the repo cache.

**Comment by azinneera** [comment]
I think we need to modify the exception message in `BallerinaUserHome.readSettings()` method. This is the place that the Settings.toml is created.
 https://github.com/ballerina-platform/ballerina-lang/blob/master/compiler/ballerina-lang/src/main/java/io/ballerina/projects/internal/environment/BallerinaUserHome.java#L79-L85

File: compiler/ballerina-lang/src/main/java/io/ballerina/projects/internal/environment/BallerinaUserHome.java [hunk 3]
```diff
@@ -79,10 +83,12 @@ private Settings readSettings() {
         if (Files.notExists(settingsFilePath)) {
             try {
                 Files.createFile(settingsFilePath);
+            } catch (AccessDeniedException ae) {
+                throw new ProjectException("permission denied to create the file: "
+                        + ProjectConstants.SETTINGS_FILE_NAME);
```
Comment (azinneera) [inline compiler/ballerina-lang/src/main/java/io/ballerina/projects/internal/environment/BallerinaUserHome.java:88]:
```suggestion
                        + ProjectConstants.SETTINGS_FILE_NAME + " in " 
                         + this.ballerinaUserHomeDirPath);
```

**Review by azinneera — APPROVED** [review]

**Pull request merged**